packages/proxy/lib/http/util/prerequests.ts

```diff
@@ -186,12 +186,24 @@ export class PreRequests {
     const pendingRequest = this.pendingRequests.shift(key)
 
     if (pendingRequest) {
+      let cdpLagDuration; let proxyRequestCorrelationDuration = 0
+
+      if (browserPreRequest.cdpRequestWillBeSentReceivedTimestamp) {
+        if (browserPreRequest.cdpRequestWillBeSentTimestamp) {
+          cdpLagDuration = browserPreRequest.cdpRequestWillBeSentReceivedTimestamp - browserPreRequest.cdpRequestWillBeSentTimestamp
+        }
+
+        if (pendingRequest.proxyRequestReceivedTimestamp) {
+          proxyRequestCorrelationDuration = Math.max(browserPreRequest.cdpRequestWillBeSentReceivedTimestamp - pendingRequest.proxyRequestReceivedTimestamp, 0)
+        }
+      }
+
       const timings = {
-        cdpRequestWillBeSentTimestamp: browserPreRequest.cdpRequestWillBeSentTimestamp,
-        cdpRequestWillBeSentReceivedTimestamp: browserPreRequest.cdpRequestWillBeSentReceivedTimestamp,
+        cdpRequestWillBeSentTimestamp: browserPreRequest.cdpRequestWillBeSentTimestamp ?? 0,
```

Comment:
since we don't get these values with BiDi I am currently coalescing them. @ryanthemanuel do you remember exactly what we are using these timings for again?

Reply:
We were using them to track on some data in protocol.

Reply:
since we don't send firefox through protocol, is this coalescing of concern or do you think we should handle the typings better here? It does feel a little awkward

Reply:
No I think that's fine.